### ISSUE TITLE
Investigate if kernel function tracing with arguments is possible

### ISSUE THREAD

**Issue by honggyukim** [namhyung/uftrace#1754]
In Linux kernel, we have `HAVE_DYNAMIC_FTRACE_WITH_ARGS` config now at https://github.com/torvalds/linux/blob/v6.4/kernel/trace/Kconfig#L48-L55.

It looks it makes kernel function tracing with function arguments so we need to investigate if we can integrate this feature so that uftrace can record and replay kernel function arguments just like we do in user space functions.

There are some references as follows.
- https://lpc.events/event/11/contributions/1106/attachments/761/1431/lpc-2021-ftrace-args.pdf
- https://lore.kernel.org/all/20201028131909.738751907@goodmis.org/T/#m54a0112cf09e3895d6275cf497378c811dbf1a0c

**Comment by namhyung** [comment]
Yep, I think we need a BTF parser and a way to pass the argument info to ftrace.  I haven't checked `fprobe` yet, but it might have the ability.
